### PR TITLE
Permissions service

### LINKED ISSUE #2791: Introduce (and mock) the IPermissionService
Ref #2754 for architectural details
Ref #2757 for API usage details

We need to introduce the `IPermissionService` interface, allow for it to be registered from the Foundation initialization and mock it in the Foundation layer so that Droid (and iOS for the time being) don't crash, similar to what we're doing with the PasswordManagerService.

Tests should be added as applicable.

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?

Introduces a new `IPermissionsService` as a central point to request and query permissions.

### Relationships

Closes #2791 

## Why do we want this?

To request permissions and query authorization status for services like calendar and notifications.

## How is it done?

It's a new service for both platforms, Android for the moment has a dummy implementation that does nothing.

### Why not another way?

It's not like we have a lot of options here 😛

### Side effects

None.

## Review hints

To test, resolve the service from the main view controller (or any other place), subscribe to the observable and request authorization.

## :squid: Permissions

Only member of the fellowship of the calendar.

## WIP

Still one thing pending, we need to check the permissions again when returning from the settings app.

### CODE REVIEW

**Comment by daividssilverio** [comment]
🤔I thought this had some Android stuff to review :/

File: Toggl.Giskard/Toggl.Giskard.csproj [hunk 1]
```diff
@@ -262,6 +266,7 @@
     <Compile Include="Resources\Resource.designer.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="LinkerPleaseInclude.cs" />
+    <Compile Include="Services\PermissionsService.cs" />
```
Comment (daividssilverio) [inline Toggl.Giskard/Toggl.Giskard.csproj]: Can you remove the changes from all the other lines? this is the only real change here.

File: Toggl.Daneel/Services/PermissionsService.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        #endregion
+
+        public void EnterForeground()
+        {
+            calendarAuthorizationStatusSubject.OnNext(isCalendarAuthorized());
+        }
+
+        void openAppSettings()
```
Comment (willsb) [inline Toggl.Daneel/Services/PermissionsService.cs]: private

File: Toggl.Daneel/Startup/Setup.cs [hunk 3]
```diff
@@ -106,7 +106,6 @@ protected override void InitializeApp(IMvxPluginManager pluginManager, IMvxAppli
                     .WithBackgroundService(new BackgroundService(timeService))
                     .WithApplicationShortcutCreator<ApplicationShortcutCreator>()
                     .WithSuggestionProviderContainer(suggestionProviderContainer)
-
```
Comment (willsb) [inline Toggl.Daneel/Startup/Setup.cs]: This line was there on purpose to separate foundation from MvvmCross

File: Toggl.Daneel/Startup/Setup.cs [hunk 4]
```diff
@@ -118,6 +117,7 @@ protected override void InitializeApp(IMvxPluginManager pluginManager, IMvxAppli
                     .WithAccessRestrictionStorage(settingsStorage)
                     .WithPasswordManagerService<OnePasswordService>()
                     .WithErrorHandlingService(new ErrorHandlingService(navigationService, settingsStorage))
+                    .WithPermissionsService(new PermissionsService())
```
Comment (willsb) [inline Toggl.Daneel/Startup/Setup.cs]: Add the extension that allows us to use the generic constructor (and don't forget to `[Preserve]` the PermissionsService)

File: Toggl.Foundation.MvvmCross/Services/IPermissionsService.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Toggl.Foundation.MvvmCross.Services
+{
+    public interface IPermissionsService
+    {
+        IObservable<bool> CalendarAuthorizationStatus { get; }
+
+        void RequestCalendarAuthorization(bool force = false);
+
+        void EnterForeground();
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/Services/IPermissionsService.cs]: Not sure if this is needed for Android, but we can always delete it later.

File: Toggl.Foundation.Tests/TogglFoundationTests.cs [hunk 6]
```diff
@@ -11,6 +11,7 @@
 using Toggl.Foundation.Analytics;
 using System.Reactive.Concurrency;
 using Toggl.Ultrawave.Network;
+using Toggl.Foundation.MvvmCross.Services;
```
Comment (willsb) [inline Toggl.Foundation.Tests/TogglFoundationTests.cs]: Is this change needed?

File: Toggl.Giskard/Startup/Setup.cs [hunk 7]
```diff
@@ -96,7 +96,6 @@ protected override void InitializeApp(IMvxPluginManager pluginManager, IMvxAppli
                     .WithBackgroundService(new BackgroundService(timeService))
                     .WithSuggestionProviderContainer(suggestionProviderContainer)
                     .WithApplicationShortcutCreator(new ApplicationShortcutCreator(ApplicationContext))
-
```
Comment (willsb) [inline Toggl.Giskard/Startup/Setup.cs]: Similar to the comment in the previous Setup.cs file

File: Toggl.Giskard/Toggl.Giskard.csproj [hunk 8]
```diff
@@ -270,8 +271,8 @@
     <Compile Include="Activities\OnboardingActivity.cs" />
     <Compile Include="Activities\LoginActivity.cs" />
     <Compile Include="Activities\LoginActivity.ViewBinds.cs">
-		<DependentUpon>LoginActivity.cs</DependentUpon>
-	</Compile>
+			<DependentUpon>LoginActivity.cs</DependentUpon>
```
Comment (willsb) [inline Toggl.Giskard/Toggl.Giskard.csproj]: pls remove

File: Toggl.Foundation.MvvmCross/Services/IPermissionsService.cs (hunk 5 above)
Comment (juanlaube) [reply Toggl.Foundation.MvvmCross/Services/IPermissionsService.cs]: Probably not, but without this in the interface, a cast is required in the AppDelegate to call this method and that's ugly
